### ISSUE TITLE
LTE detected should support all tested compatible LTE modules

### ISSUE THREAD

**Issue by shawaj** [NebraLtd/hm-diag#88]
See list of tested units here:

https://github.com/NebraLtd/helium-miner-software/wiki/3G---4G-Module-Compatability-List

Only currently looks for 2c7c in USB vendor ID

https://github.com/NebraLtd/hm-diag/blob/1444a3b2ccfcec7c6b505ede8dd2ecfa31b1e831/diagnostics-program/main.py#L91

**Comment by shawaj** [comment]
Perhaps it should instead use `mmcli -L` or similar to detect a working modem instead?

**Comment by shawaj** [comment]
EG25G-S = ID 2c7c:0125 Quectel Wireless Solutions Co., Ltd. EC25 LTE modem
EG25G = ID 2c7c:0125 Quectel Wireless Solutions Co., Ltd. EC25 LTE modem
sim7100e = ID 1e0e:9001 Qualcomm / Option
reyax ryx9000 = ID 1bc7:1201 Telit Wireless Solutions
ec25-e = ID 2c7c:0125 Quectel Wireless Solutions Co., Ltd. EC25 LTE modem
eg21-g = ID 2c7c:0121 Quectel Wireless Solutions Co., Ltd. EC21 LTE modem
BG95-M3 = ID 2c7c:0700 Quectel Wireless Solutions Co., Ltd.
BG96 = ID 2c7c:0296 Quectel Wireless Solutions Co., Ltd. BG96 CAT-M1/NB-IoT modem
Sierra mc7700 = ID 1199:68a2 Sierra Wireless, Inc.

**Comment by shawaj** [comment]
See also table at https://github.com/NebraLtd/Helium-Guides/blob/main/docs/outdoor-hotspot/lte-config.md

**Comment by shawaj** [comment]
Huawei ME909s-120 = ID 12d1:15c1 Huawei Technologies Co., Ltd. ME906s LTE M.2 Module
```
root@c03ac42:~# mmcli -m 0
  -----------------------------
  General  |         dbus path: /org/freedesktop/ModemManager1/Modem/0
           |         device id: cbcec58bece9c9fc9b2fb5c338dbca65d2c3f715
  -----------------------------
  Hardware |      manufacturer: Huawei Technologies Co., Ltd.
           |             model: HUAWEI Mobile V7R11
           | firmware revision: 11.617.01.00.00
           |      h/w revision: RM1ME909ASM
           |         supported: gsm-umts, lte
           |           current: gsm-umts, lte
           |      equipment id: 867377025718697
  -----------------------------
  System   |            device: /sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2
           |           drivers: cdc_mbim
           |            plugin: huawei
           |      primary port: cdc-wdm0
           |             ports: cdc-wdm0 (mbim), wwan0 (net)
  -----------------------------
  Status   |             state: failed
           |     failed reason: sim-missing
           |       power state: on
           |    signal quality: 0% (cached)
  -----------------------------
  Modes    |         supported: allowed: 2g, 3g, 4g; preferred: none
           |           current: allowed: any; preferred: none
  -----------------------------
  IP       |         supported: ipv4, ipv6, ipv4v6
```

**Comment by shawaj** [comment]
Sierra mc7700 = ID 1199:68a2 Sierra Wireless, Inc.

root@c03ac42:~# mmcli -m 0
error: couldn't find modem

**Comment by shawaj** [comment]
MikroTik R11e-LTE6 = ID 2cd2:0004
```
root@c03ac42:~# mmcli -m 0
error: couldn't find modem
```

**Comment by shawaj** [comment]
sim7100e = ID 1e0e:9001 Qualcomm / Option
```
root@c03ac42:~# mmcli -m 3
  -----------------------------
  General  |         dbus path: /org/freedesktop/ModemManager1/Modem/3
           |         device id: da87fd78b247b9849a4a05bf663228bba0a6a0db
  -----------------------------
  Hardware |      manufacturer: QUALCOMM INCORPORATED
           |             model: SIMCOM_SIM7100E
           | firmware revision: M9615A-CETWTBZM-6.0.15255  1  [May 13 2016 01:00:00]
           |      h/w revision: 30000
           |         supported: gsm-umts, lte
           |           current: gsm-umts, lte
           |      equipment id: 866802021636586
  -----------------------------
  System   |            device: /sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2
           |           drivers: option1, qmi_wwan
           |            plugin: simtech
           |      primary port: cdc-wdm0
           |             ports: cdc-wdm0 (qmi), ttyUSB0 (qcdm), ttyUSB2 (at), wwan0 (net), 
           |                    ttyUSB4 (audio), ttyUSB1 (gps), ttyUSB3 (at)
  -----------------------------
  Status   |             state: failed
           |     failed reason: sim-missing
           |       power state: on
           |    signal quality: 0% (cached)
  -----------------------------
  Modes    |         supported: allowed: 2g; preferred: none
           |                    allowed: 3g; preferred: none
           |                    allowed: 4g; preferred: none
           |                    allowed: 2g, 3g; preferred: 3g
           |                    allowed: 2g, 3g; preferred: 2g
           |                    allowed: 2g, 4g; preferred: 4g
           |                    allowed: 2g, 4g; preferred: 2g
           |                    allowed: 3g, 4g; preferred: 4g
           |                    allowed: 3g, 4g; preferred: 3g
           |                    allowed: 2g, 3g, 4g; preferred: 4g
           |                    allowed: 2g, 3g, 4g; preferred: 3g
           |                    allowed: 2g, 3g, 4g; preferred: 2g
           |           current: allowed: any; preferred: none
  -----------------------------
  Bands    |         supported: egsm, dcs, utran-1, utran-3, utran-5, utran-8, eutran-1, 
           |                    eutran-3, eutran-7, eutran-8, eutran-20, eutran-38, eutran-40
  -----------------------------
  IP       |         supported: ipv4, ipv6, ipv4v6
```

**Comment by shawaj** [comment]
reyax ryx9000 = ID 1bc7:1201 Telit Wireless Solutions
```
root@c03ac42:~# mmcli -m 6
  -----------------------------
  General  |         dbus path: /org/freedesktop/ModemManager1/Modem/6
           |         device id: df6a4b9224e6144271db2b66a87ffa40354fa833
  -----------------------------
  Hardware |      manufacturer: Telit
           |             model: LE910C4-EU
           | firmware revision: 25.20.678  1  [May 12 2020 00:00:00]
           |    carrier config: default
           |      h/w revision: 1.10
           |         supported: gsm-umts, lte
           |           current: gsm-umts, lte
           |      equipment id: 351626102534010
  -----------------------------
  System   |            device: /sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2
           |           drivers: qmi_wwan, option1
           |            plugin: telit
           |      primary port: cdc-wdm0
           |             ports: cdc-wdm0 (qmi), ttyUSB0 (unknown), ttyUSB2 (at), 
           |                    ttyUSB4 (unknown), wwan0 (net), ttyUSB1 (gps), ttyUSB3 (at)
  -----------------------------
  Status   |             state: failed
           |     failed reason: sim-missing
           |       power state: on
           |    signal quality: 0% (cached)
  -----------------------------
  Modes    |         supported: allowed: 2g; preferred: none
           |                    allowed: 3g; preferred: none
           |                    allowed: 4g; preferred: none
           |                    allowed: 2g, 3g; preferred: 3g
           |                    allowed: 2g, 3g; preferred: 2g
           |                    allowed: 2g, 4g; preferred: 4g
           |                    allowed: 2g, 4g; preferred: 2g
           |                    allowed: 3g, 4g; preferred: 4g
           |                    allowed: 3g, 4g; preferred: 3g
           |                    allowed: 2g, 3g, 4g; preferred: 4g
           |                    allowed: 2g, 3g, 4g; preferred: 3g
           |                    allowed: 2g, 3g, 4g; preferred: 2g
           |           current: allowed: any; preferred: none
  -----------------------------
  Bands    |         supported: egsm, dcs, utran-1, utran-3, utran-8, eutran-1, eutran-3, 
           |                    eutran-7, eutran-8, eutran-20, eutran-28
  -----------------------------
  IP       |         supported: ipv4, ipv6, ipv4v6
```

**Comment by shawaj** [comment]
EG25G-S = ID 2c7c:0125 Quectel Wireless Solutions Co., Ltd. EC25 LTE modem
EG25G = ID 2c7c:0125 Quectel Wireless Solutions Co., Ltd. EC25 LTE modem
```
root@c03ac42:~# mmcli -m 7
  -----------------------------------
  General  |               dbus path: /org/freedesktop/ModemManager1/Modem/7
           |               device id: 702f6b8507808aa397db70ff62f718078f192531
  -----------------------------------
  Hardware |            manufacturer: QUALCOMM INCORPORATED
           |                   model: QUECTEL Mobile Broadband Module
           |       firmware revision: EG25GGBR07A08M2G
           |          carrier config: ROW_Generic_3GPP
           | carrier config revision: 0501081F
           |            h/w revision: 10000
           |               supported: gsm-umts, lte
           |                 current: gsm-umts, lte
           |            equipment id: 867698043007338
  -----------------------------------
  System   |                  device: /sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2
           |                 drivers: qmi_wwan, option1
           |                  plugin: quectel
           |            primary port: cdc-wdm0
           |                   ports: cdc-wdm0 (qmi), ttyUSB0 (qcdm), ttyUSB2 (at), wwan0 (net), 
           |                          ttyUSB1 (gps), ttyUSB3 (at)
  -----------------------------------
  Status   |                   state: failed
           |           failed reason: sim-missing
           |             power state: on
           |          signal quality: 0% (cached)
  -----------------------------------
  Modes    |               supported: allowed: 2g; preferred: none
           |                          allowed: 3g; preferred: none
           |                          allowed: 4g; preferred: none
           |                          allowed: 2g, 3g; preferred: 3g
           |                          allowed: 2g, 3g; preferred: 2g
           |                          allowed: 2g, 4g; preferred: 4g
           |                          allowed: 2g, 4g; preferred: 2g
           |                          allowed: 3g, 4g; preferred: 4g
           |                          allowed: 3g, 4g; preferred: 3g
           |                          allowed: 2g, 3g, 4g; preferred: 4g
           |                          allowed: 2g, 3g, 4g; preferred: 3g
           |                          allowed: 2g, 3g, 4g; preferred: 2g
           |                 current: allowed: any; preferred: none
  -----------------------------------
  Bands    |               supported: egsm, dcs, pcs, g850, utran-1, utran-4, utran-6, utran-5, 
           |                          utran-8, utran-2, eutran-1, eutran-2, eutran-3, eutran-4, eutran-5, 
           |                          eutran-7, eutran-8, eutran-12, eutran-13, eutran-18, eutran-19, 
           |                          eutran-20, eutran-25, eutran-26, eutran-28, eutran-38, eutran-39, 
           |                          eutran-40, eutran-41, utran-19
  -----------------------------------
  IP       |               supported: ipv4, ipv6, ipv4v6
```

**Comment by shawaj** [comment]
BG96 = ID 2c7c:0296 Quectel Wireless Solutions Co., Ltd. BG96 CAT-M1/NB-IoT modem
```
root@c03ac42:~# mmcli -m 8
  -----------------------------
  General  |         dbus path: /org/freedesktop/ModemManager1/Modem/8
           |         device id: 7336675dc556d5cbd1828788b1cedb83e0497392
  -----------------------------
  Hardware |      manufacturer: Quectel
           |             model: 0
           | firmware revision: BG96MAR02A07M1G
           |    carrier config: default
           |      h/w revision: 10000
           |         supported: gsm-umts, lte
           |           current: gsm-umts, lte
           |      equipment id: 864431044319483
  -----------------------------
  System   |            device: /sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2
           |           drivers: qmi_wwan, option1
           |            plugin: quectel
           |      primary port: cdc-wdm0
           |             ports: cdc-wdm0 (qmi), ttyUSB0 (qcdm), ttyUSB2 (at), wwan0 (net), 
           |                    ttyUSB1 (gps), ttyUSB3 (at)
  -----------------------------
  Status   |             state: failed
           |     failed reason: sim-missing
           |       power state: on
           |    signal quality: 0% (cached)
  -----------------------------
  Modes    |         supported: allowed: 2g; preferred: none
           |                    allowed: 4g; preferred: none
           |                    allowed: 2g, 4g; preferred: 4g
           |                    allowed: 2g, 4g; preferred: 2g
           |           current: allowed: any; preferred: none
  -----------------------------
  Bands    |         supported: egsm, dcs, pcs, g850, eutran-1, eutran-2, eutran-3, 
           |                    eutran-4, eutran-5, eutran-8, eutran-12, eutran-13, eutran-17, 
           |                    eutran-18, eutran-19, eutran-20, eutran-26, eutran-28, eutran-39
  -----------------------------
  IP       |         supported: ipv4, ipv6, ipv4v6
```

**Comment by shawaj** [comment]
eg21-g = ID 2c7c:0121 Quectel Wireless Solutions Co., Ltd. EC21 LTE modem
```
root@c03ac42:~# mmcli -m 9
  -----------------------------------
  General  |               dbus path: /org/freedesktop/ModemManager1/Modem/9
           |               device id: e3dd9d6271f145e3fb375f5b656f7c56401fe4fd
  -----------------------------------
  Hardware |            manufacturer: QUALCOMM INCORPORATED
           |                   model: QUECTEL Mobile Broadband Module
           |       firmware revision: EG21GGBR07A11M1G
           |          carrier config: ROW_Generic_3GPP
           | carrier config revision: 05040820
           |            h/w revision: 10000
           |               supported: gsm-umts, lte
           |                 current: gsm-umts, lte
           |            equipment id: 862771040196662
  -----------------------------------
  System   |                  device: /sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2
           |                 drivers: qmi_wwan, option1
           |                  plugin: quectel
           |            primary port: cdc-wdm0
           |                   ports: cdc-wdm0 (qmi), ttyUSB0 (qcdm), ttyUSB2 (at), wwan0 (net), 
           |                          ttyUSB3 (at)
  -----------------------------------
  Status   |                   state: failed
           |           failed reason: sim-missing
           |             power state: on
           |          signal quality: 0% (cached)
  -----------------------------------
  Modes    |               supported: allowed: 2g; preferred: none
           |                          allowed: 3g; preferred: none
           |                          allowed: 4g; preferred: none
           |                          allowed: 2g, 3g; preferred: 3g
           |                          allowed: 2g, 3g; preferred: 2g
           |                          allowed: 2g, 4g; preferred: 4g
           |                          allowed: 2g, 4g; preferred: 2g
           |                          allowed: 3g, 4g; preferred: 4g
           |                          allowed: 3g, 4g; preferred: 3g
           |                          allowed: 2g, 3g, 4g; preferred: 4g
           |                          allowed: 2g, 3g, 4g; preferred: 3g
           |                          allowed: 2g, 3g, 4g; preferred: 2g
           |                 current: allowed: any; preferred: none
  -----------------------------------
  Bands    |               supported: egsm, dcs, pcs, g850, utran-1, utran-4, utran-6, utran-5, 
           |                          utran-8, utran-2, eutran-1, eutran-2, eutran-3, eutran-4, eutran-5, 
           |                          eutran-7, eutran-8, eutran-12, eutran-13, eutran-18, eutran-19, 
           |                          eutran-20, eutran-25, eutran-26, eutran-28, eutran-38, eutran-39, 
           |                          eutran-40, eutran-41, utran-19
  -----------------------------------
  IP       |               supported: ipv4, ipv6, ipv4v6
```

**Comment by shawaj** [comment]
BG95-M3 = ID 2c7c:0700 Quectel Wireless Solutions Co., Ltd.
```
root@c03ac42:~# mmcli -L
No modems were found
```

**Comment by shawaj** [comment]
ec25-e = ID 2c7c:0125 Quectel Wireless Solutions Co., Ltd. EC25 LTE modem
```
root@c03ac42:~# mmcli -m 10
  -----------------------------
  General  |         dbus path: /org/freedesktop/ModemManager1/Modem/10
           |         device id: a8858af7f03227a6671538dfa4c52ee6eb276586
  -----------------------------
  Hardware |      manufacturer: QUALCOMM INCORPORATED
           |             model: QUECTEL Mobile Broadband Module
           | firmware revision: EC25EFAR06A02M4G
           |    carrier config: default
           |      h/w revision: 10000
           |         supported: gsm-umts, lte
           |           current: gsm-umts, lte
           |      equipment id: 866758042149852
  -----------------------------
  System   |            device: /sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2
           |           drivers: qmi_wwan, option1
           |            plugin: quectel
           |      primary port: cdc-wdm0
           |             ports: cdc-wdm0 (qmi), ttyUSB0 (qcdm), ttyUSB2 (at), wwan0 (net), 
           |                    ttyUSB1 (gps), ttyUSB3 (at)
  -----------------------------
  Status   |             state: failed
           |     failed reason: sim-missing
           |       power state: on
           |    signal quality: 0% (cached)
  -----------------------------
  Modes    |         supported: allowed: 2g; preferred: none
           |                    allowed: 3g; preferred: none
           |                    allowed: 4g; preferred: none
           |                    allowed: 2g, 3g; preferred: 3g
           |                    allowed: 2g, 3g; preferred: 2g
           |                    allowed: 2g, 4g; preferred: 4g
           |                    allowed: 2g, 4g; preferred: 2g
           |                    allowed: 3g, 4g; preferred: 4g
           |                    allowed: 3g, 4g; preferred: 3g
           |                    allowed: 2g, 3g, 4g; preferred: 4g
           |                    allowed: 2g, 3g, 4g; preferred: 3g
           |                    allowed: 2g, 3g, 4g; preferred: 2g
           |           current: allowed: any; preferred: none
  -----------------------------
  Bands    |         supported: egsm, dcs, utran-1, utran-5, utran-8, eutran-1, eutran-3, 
           |                    eutran-5, eutran-7, eutran-8, eutran-20, eutran-38, eutran-40, 
           |                    eutran-41
  -----------------------------
  IP       |         supported: ipv4, ipv6, ipv4v6
```

**Comment by vpetersson** [comment]
> Perhaps it should instead use `mmcli -L` or similar to detect a working modem instead?

Nice findings! So we shouldn't really "shell out" to something like `nmcli` but rather integrate with Network Manager directly over DBUS. That will be far more reliable.

**Comment by shawaj** [comment]
@marvinmarnold @vpetersson need to get this resolved in order for diagnostics to work properly with RockPi

**Comment by marvinmarnold** [comment]
Thanks for calling this out. I'll need to catch up on the details. Marked it for next sprint.

**Comment by marvinmarnold** [comment]
@posterzh can you confirm that this is resolved by #280 and close if that's the case.

**Comment by marvinmarnold** [comment]
Closed by https://github.com/NebraLtd/helium-miner-software/pull/280
